src/Npgsql/Internal/TypeInfoMapping.cs

```diff
@@ -652,18 +653,33 @@ void AddPolymorphicResolverArrayType(TypeInfoMapping elementMapping, Type type,
     }
 
     /// Returns whether type matches any of the types we register pg arrays as.
+    [UnconditionalSuppressMessage("Trimming", "IL2070",
+        Justification = "We care about IList<T> implementations if the instantiation is actually rooted by us through a registered mapping.")]
```

Comment:
Possibly add just a bit more info that the rooting happens in GetListTypeMatchPredicate() and won't happen in the dynamic case etc. (for future self)

Reply:
Sure, will add some more context